### DISCUSSION TITLE
Twilio login question

### DISCUSSION THREAD

**Discussion by KULTI1995** [supabase/supabase#2827] [Questions] [answered]
https://supabase.io/docs/guides/auth/auth-twilio

How to make a user constantly logged in? Login similar to Whatsapp. You log in with your phone once, enter a code and the login persists. Because from the documentation it seems that the user will have to log in every time....

**Comment by dshukertjr** [comment, marked as answer]
Hi @KULTI1995!

This might be related to https://github.com/supabase/gotrue/issues/141

Right now, the gotrue server does not return `user` on successful OTP verification. One this issue is resolved, you should be able to persist your auth state!

**Reply by KULTI1995** [reply]
Okey, thanks.
